src/components/Context/createComponent.js

```diff
@@ -28,8 +28,9 @@ export default (config, logger, availableContexts, requiredContexts) => {
     lifecycle: {
       onBeforeEvent({ event }) {
         const xdm = {};
-        contexts.forEach(context => context(xdm));
-        event.mergeXdm(xdm);
+        return Promise.all(
```

Comment:
we need this to collect all the contexts data?

Reply:
I think he needed to wrap it with a promise because the Google API is async

Reply:
That's correct